crates/neon-sys/native/src/neon.cc

```diff
@@ -221,15 +221,21 @@ extern "C" bool Neon_ArrayBuffer_New(v8::Local<v8::ArrayBuffer> *out, v8::Isolat
 
 
 extern "C" size_t Neon_ArrayBuffer_Data(v8::Isolate *isolate, void **base_out, v8::Local<v8::ArrayBuffer> buffer) {
-#if (V8_MAJOR_VERSION >= 8)
-    auto contents = buffer->GetBackingStore();
-    *base_out = contents->Data();
-    return contents->ByteLength();
-#else
-    v8::ArrayBuffer::Contents contents = buffer->GetContents();
-    *base_out = contents.Data();
-    return contents.ByteLength();
-#endif
+  #if _MSC_VER && NODE_RUNTIME_ELECTRON && NODE_MODULE_VERSION >= 89
```

Comment:
If there's no way to reliably detect we're in electron (since `NODE_RUNTIME_ELECTRON` is only set by `electron-build-env` and so not super reliable), we can just unconditionally use this hack for all Windows builds with the legacy backend. It's probably a performance hit but only for the legacy backend, and we'll encourage people to upgrade to the new backend anyway.